config/routes.rb

```diff
@@ -283,6 +283,10 @@
       get 'extlogin'
       get 'extlogout'
       get 'auto_complete_search'
+      get 'stop_impersonation'
```

Comment:
Is it necessary to be `get`?
As both actions change state of the app, it would feel more natural to use:
- `delete` for `stop_impersonation`,
- `post` for `impersonate`.

I can see how it could make your life difficult in the React part though... :sob:

Reply:
I think it is `get` for simplicity and convenience to trigger rails routing redirect from js code. I swapped both as you suggested. Changing `impersonate` to `post` was not a problem, however for `stop_impersonation`, I am now unable to show toast after the redirect and it is shown only before the page refreshes.

So both versions are far from perfect, maybe we can go half-way and have `impersonate` as `post` but keep `stop_impersonation` as `get`, depends on what you prefer.

Reply:
I :heart: you have manage to do both! :+1: